static/app/views/dashboards/widgetCard/autoSizedText.tsx

```diff
@@ -0,0 +1,143 @@
+import {useLayoutEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import * as Sentry from '@sentry/react';
+
+interface Props {
+  children: React.ReactNode;
+  maximumDifference?: number;
+}
+
+export function AutoSizedText({
+  children,
+  maximumDifference = DEFAULT_MAXIMUM_DIFFERENCE,
+}: Props) {
+  const childRef = useRef<HTMLDivElement>(null);
+
+  const fontSize = useRef<number>(0);
+  const fontSizeLowerBound = useRef<number>(0);
+  const fontSizeUpperBound = useRef<number>(0);
+
+  useLayoutEffect(() => {
+    const childElement = childRef.current; // This is `SizedChild`
+    const parentElement = childRef.current?.parentElement; // This is the parent of `AutoSizedText`
+
+    if (!childElement || !parentElement) {
+      return () => {};
```

Comment:
iirc that consistent return rule doesnt actually care about the type
```suggestion
      return undefined;
```